packages/flutter_tools/test/src/context.dart

```diff
@@ -81,6 +81,7 @@ void testUsingContext(String description, dynamic testMethod(), {
           SimControl: () => MockSimControl(),
           Usage: () => MockUsage(),
           XcodeProjectInterpreter: () => MockXcodeProjectInterpreter(),
+          FileSystem: () => LocalFileSystemBlockingSetCurrentDirectory(),
```

Comment:
I'm curious what would happen if we used something more aggressive here - like a file system that threw for all operations, forcing the test author to always provide a test-specific file system.

Reply:
I think it'd add quite a bit of boilerplate for a relatively small benefit. Most use of the file system is safe, it's only really global mutable state like this causing issues. It'd be nice if we could default to an in-memory file system here, but we have many tests that require reading/writing real files (for ex. they write files into temp folder, then spawn processes that interact with them).